bot/interact.go

```diff
@@ -128,6 +112,49 @@ func interactAPIHandler(botContext cmd.BotContext) http.HandlerFunc {
 	}
 }
 
+// Runs validations over the submitted salary range and job offer link. Produces a list of errors if any.
+//
+// Arguments are the strings as read from the submissions (no previous transform/parsing/filter)
+// Returns:
+//  - the parsed max salary as int
+//  - the parsed min salary as int, or -1 if the field was empty (it's an optional field)
+//  - a map of field name to error message
+func ValidateSubmission(messageJobLink string, messageMaxSalary string, messageMinSalary string) (int, int, map[string]string) {
```

Comment:
- don't need to export it
- little trick on args of the same type
```suggestion
func validateSubmission(messageJobLink, messageMaxSalary, messageMinSalary string) (int, int, map[string]string) {
```

I'd maybe rename it to `validateSalary`, or keep the name and pass the whole submission instead of assigning the args + passing them.

Reply:
I read from the submission and passed the strings so it was simpler to write the test.